app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java

```diff
@@ -171,34 +165,99 @@ public void removeUser(View view) {
-                 .addOnCompleteListener(task -> {
-                     Map<String, Object> householdData = task.getResult().getData();
-                     if(householdData != null) {
-                         List<String> residents =
-                                 (List<String>) householdData.getOrDefault("residents", "[]");
-                         Long num_users = (Long) householdData.get("num_members");
-                         if(residents.contains(email)){
-                             currentHousehold.update("num_members",num_users-1);
-                             currentHousehold.update("residents", FieldValue.arrayRemove(email));
-                             Toast.makeText(getApplicationContext(),
-                                     view.getContext().getString(R.string.remove_user_success),
-                                     Toast.LENGTH_SHORT).show();
-                         }
-                     }
-                 });
+    public void deleteDialog(View view) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("The household calendar and lists will be deleted too. " +
+                "Are you sure you want to delete this household?");
+        builder.setTitle("Delete household");
+        builder.setCancelable(false);
+        builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+        builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int which) {
+                deleteCalendar(view);
+                deleteGroceryList(view);
+                deleteTaskList(view);
+                deleteHousehold(view);
+                dialog.cancel();
+            }
+        });
+
+        AlertDialog alert = builder.create();
+        alert.show();
+    }
+
+    public void deleteGroceryList(View view) {
+        // TODO : The grocery list is not linked to households yet
+    }
+
+    public void deleteTaskList(View view) {
+        // TODO : The task list is not linked to households yet
+    }
+
+    public void deleteCalendar(View view) {
+        firestore.collection("events")
+                .whereEqualTo("household", currentHousehold)
+                .get().addOnCompleteListener(task1 -> {
+            if (task1.isSuccessful()) {
+                for (QueryDocumentSnapshot document : task1.getResult()) {
+                    firestore.collection("events").document(document.getId()).delete()
+                            .addOnCompleteListener(task2 -> {
+                                if(!task2.isSuccessful()) {
+                                    Toast.makeText(getApplicationContext(), view.getContext().getString(R.string.remove_calendar_failure), Toast.LENGTH_SHORT).show();
+                                }
+                            });
```

Comment:
[OPTIONAL] Instead of using two onCompleteListeners you can use continueWithTask, return a taask and then put an onComplete listener on everything
I spent quite som etime figuring out how it works so if you want you can check MainScreenActivity in the branch groceries_refactor